packages/widgetbook/lib/src/knobs/knobs.dart

```diff
@@ -0,0 +1,79 @@
+import 'package:flutter/widgets.dart';
+import 'package:provider/provider.dart';
+import 'package:widgetbook/src/knobs/bool_knob.dart';
+import 'package:widgetbook/src/repositories/selected_story_repository.dart';
+
+import 'knobs_builder.dart';
+
+abstract class Knob<T> {
+  Knob({
+    required this.label,
+    this.description,
+    required this.value,
+  });
+
+  /// This is the current value the knob is set to
+  T value;
+
+  /// This is a description the user can put on the knob
+  final String? description;
+
+  /// This is the label that's put above a knob
+  final String label;
+
+  Widget build();
+}
+
+extension Knobs on BuildContext {
+  KnobsBuilder get knobs => watch<KnobsNotifier>();
+}
+
+class KnobsNotifier extends ChangeNotifier implements KnobsBuilder {
+  KnobsNotifier(this._selectedStoryRepository) {
+    _selectedStoryRepository.getStream().listen((event) => notifyListeners());
+  }
```

Comment:
So that the knobs can be reselected, when we dont the knobs will never be reconstructed. You can comment out this line and see that the knobs wont update when switching use cases